### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by LightDestory**
This PR will address the following:

- Restructuring to use Pythonic-paths
- Add missing dependency on Requirements
- Add target version for depenencies on Requirements
- Remove decorators in favour of registering handlers
- Fix inefficient camera access, saving a lot of time
- Add Peek command
- Registering commands for UI friendly
- Lint/Format Pipeline

### CODE REVIEW

**Comment by LightDestory** [comment]
I don't have the required oauth credentiuals to actually test the image upload. Everything else should be working.

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                if not endless:
+                    duration -= 30
+                    if duration <= 0:
+                        print("Capture process finished")
+                        current_status = CaptureStatus.NOT_STARTED
+                        capture_thread_duration = -1
+                        break
+            # This module should trigger the capture process every 30 seconds
+            internal_counter = (internal_counter + 1) % 60
+            time.sleep(0.5)
```
Comment (Helias) [inline src/main.py:64]: potresti usare `from sleep import time` così non importi l'intera libreria ma solo sleep()

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+def capture_job(duration: int) -> None:
+    """
+    This function captures images from the camera every 30 seconds. The process will run for the specified duration.
+    If duration is 0, the process will run indefinitely
+    """
+    global current_status, capture_thread_duration
+    internal_counter: int = 0
+    endless = duration == 0
+    duration = duration * 60
```
Comment (Helias) [inline src/main.py:49]: `duration *= 60`
in termini di memoria risparmi ([source](https://stackoverflow.com/questions/41446833/what-is-the-difference-between-i-i-1-and-i-1-in-a-for-loop))

File: src/main.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+def _start_capture_process(message: Message, bot: TeleBot) -> None:
+    """
+    This function starts the capture service for the next 30 minutes
+
+    :param message: The message object
+    :param bot: The bot object
+    """
+    global current_status, capture_thread, capture_thread_duration
+    command = message.text.split()
+    if len(command) == 1 or not command[1].isdigit():
```
Comment (Helias) [inline src/main.py:79]: non so se ha senso mettere una sorta di null-check tipo `len(command) > 0`

File: src/main.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        bot.reply_to(message, "The command must be in the format:\n/capture [duration in minutes]\nIf 0 the process "
+                              "will run indefinitely")
+        return
+    if current_status != CaptureStatus.NOT_STARTED:
+        bot.reply_to(message, "Capture service is already running")
+        return
+    capture_thread_duration = int(command[1])
+    current_status = CaptureStatus.RUNNING
+    capture_thread = threading.Thread(target=capture_job, args=(capture_thread_duration,))
```
Comment (Helias) [inline src/main.py:88]: `from threading import Thread`

File: src/main.py (hunk 3 above)
Comment (LightDestory) [reply src/main.py:91]: In quel contesto ci va solo se il comando /capture viene rilevato dalla libreria telebot, ergo uno split su "/capture <la-qualsiasi>" fornirà sempre un array di almeno 1 elemento. Questo controllo è superfluo

**Comment by LightDestory** [comment]
@Helias [44f96aa](https://github.com/UniCT-ARSLab/EurobotVision/pull/5/commits/44f96aa5369c1c79850b1894524dcebf9578c645) should close #6

**Review by DefEnge — APPROVED** [review]

**Pull request merged**